chore(core): Update API Design Guide (#8155)
island-is/island.is: handbook/technical-overview/api-design-guide/rest-response.md

# REST Response

A REST API should use an appropriate HTTP Status Code and Content Type when responding to a client's request.

## Content Type

> An API can support multiple response content types. It's preferred to use a JSON content by default.
> When an API supports multiple content types the client can specify the desired content type by using the
> [`Accept` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept).

Reviewer: ```suggestion
An API can support multiple response content types. It's preferred to use a JSON content type by default.
When an API supports multiple content types the client can specify the desired content type with the
[`Accept` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept).
```

The following table shows a list of common content types:

| Content Type                                                            | Description                                                                                    |
| ----------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------- |
| [application/json](https://www.rfc-editor.org/rfc/rfc8259)              | **Default.** Content type for a JSON response body.                                            |
| [application/problem+json](https://www.rfc-editor.org/rfc/rfc7807.html) | Content type for the [Problem JSON error object](errors.md) for `4xx` and `5xx` statuse codes. |
| [application/xml](https://www.rfc-editor.org/rfc/rfc7303#section-9.1)   | Content type for a XML response body.                                                          |

> | [text/xml](https://www.rfc-editor.org/rfc/rfc7303#section-9.2)          | Same as the `application/xml` except for the type name is `'xml'`                              |

Reviewer: I don't understand the description. 🥺

Author: I updated this and added a info section below on the difference. Which is from the previous RFC3023, but I like the distinction between those two. It says that `text/xml` should be used then the content is readable by casual users, while `application/xml` should be used when it is unreadable. The later RFC 7303 does not mention this difference anymore.

{% hint style="info" %}
When more details are needed in an error response the API should use an application
defined errors and supply them in an [error object](errors.md) in the response body.
{% endhint %}

## HTTP Status Codes

> A developer is encouraged to use the range of [`HTTP Status Codes`](https://httpstatuses.org/) to give the clients
> the mot appropriate result of the request processing.

Reviewer: ```suggestion
REST APIs should use the range of [`HTTP Status Codes`](https://httpstatuses.org/) to give clients
the most appropriate result of the request processing.
```
